### PR TITLE
Add legacy_permit_unsafe_log_operations and legacy_unsafe_log_warning_interval_sec for 23.2

### LINKED ISSUE #1485: Add new configuration flags for unsafe string checks
### Describe the Issue

For the upcoming v23.2 release of Redpanda, two new configuration options have been added that effects whether certain input from the user willbe rejected or not.

### Updates to existing documentation

<https://docs.redpanda.com/docs/reference/cluster-properties/>

I believe these new options would fit in the #support-properties section.

```
---

### legacy_permit_unsafe_log_operations

If true, this flag permits the operator to use control characters within strings (such as consumer group names or user names).

This flag is only honored in clusters that have upgraded to v23.2.  New clusters starting with v23.2 and above will ignore this flag.

Note that this flag will be removed in future versions of Redpanda and no string containing control characters will be accepted at that time.

**Type**: boolean

**Default**: true

**Restart required**: no

---

### legacy_unsafe_log_warning_interval_sec

If unsafe strings are permitted, a warning will be logged at an interval specified by this option.

**Units**: seconds

**Default**: 300

**Related properties**: [legacy_permit_unsafe_log_operations](#legacy_permit_unsafe_log_operations)

```

### New feature or content gap requests

New feature.

### If new feature, in which release is it included or expected? 

v23.2

### Additional notes

Epic: https://github.com/redpanda-data/core-internal/issues/357

RFC: https://docs.google.com/document/d/17khGAnNnvrT9fb30b8SOoddkT5XtI9T7b8wxZyqdW0w/edit#heading=h.mvlqwsjv79vf

PRs:

* https://github.com/redpanda-data/redpanda/pull/9640
* https://github.com/redpanda-data/redpanda/pull/9277

### PR DESCRIPTION
**Pull request by shorgi**
Fixes #1485

### CODE REVIEW

File: versioned_docs/version-23.2/reference/cluster-properties.mdx [hunk 1]
```diff
@@ -1068,6 +1068,38 @@ Requires that an empty file named `.redpanda_data_dir` be present in the [data d
 
 ## Support properties
 
+### legacy_permit_unsafe_log_operations
+
+Flag enabling a Redpanda cluster operator to use unsafe control characters within strings such as consumer group names or user names.
+
+This flag applies only for Redpanda clusters v23.1 or earlier that have upgraded to v23.2 or later. Newly created Redpanda clusters v23.2 or later will ignore this property. It will be removed in a future version of Redpanda, after which strings containing control characters won't be accepted.
```
Comment (michael-redpanda) [inline versioned_docs/version-23.2/reference/cluster-properties.mdx:1075]: s/It will be removed/This flag will be removed/

**Review by michael-redpanda — COMMENTED** [review]
Just one comment on wording, but everything else looks good!

**Review by michael-redpanda — APPROVED** [review]

File: versioned_docs/version-23.2/reference/cluster-properties.mdx [hunk 2]
```diff
@@ -1068,6 +1068,38 @@ Requires that an empty file named `.redpanda_data_dir` be present in the [data d
 
 ## Support properties
 
+### legacy_permit_unsafe_log_operations
+
+Flag enabling a Redpanda cluster operator to use unsafe control characters within strings such as consumer group names or user names.
+
+This flag applies only for Redpanda clusters v23.1 or earlier that have upgraded to v23.2 or later. Newly created Redpanda clusters v23.2 or later will ignore this property. This flag will be removed in a future version of Redpanda, after which strings containing control characters won't be accepted.
```
Comment (Feediver1) [inline versioned_docs/version-23.2/reference/cluster-properties.mdx:1075]:
```suggestion
This flag applies only for Redpanda clusters v23.1 or earlier that have upgraded to v23.2 or later. Newly-created Redpanda clusters v23.2 or later ignore this property. 
```

**Review by Feediver1 — APPROVED** [review]
See/implement comment removing future updates. Thx.

**Pull request merged**